tests/test_start_date.py

```diff
@@ -0,0 +1,142 @@
+
+        for value in rep_values:
+            self.assertGreaterEqual(
+                strptime_to_utc(value),
+                strptime_to_utc(start_date)
+            )
+
+    def verify_replication_key_values(self, stream_name):
+        self.do_verify_replication_key_values(
+            stream_name,
+            self.first_sync_records,
+            self.first_sync_start_date
+        )
+        self.do_verify_replication_key_values(
+            stream_name,
+            self.second_sync_records,
+            self.second_sync_start_date
+        )
+
+    def test_run(self):
+        """
+        Verify that running a sync, then moving the start date into the
+        future and running a sync results in less records than the first
+        sync
+        """
+        (_,
+         first_sync_record_count_by_stream,
+         first_sync_state,
+         first_sync_all_records_by_stream) = self.make_connection_and_run_sync()
+
+        (_,
+         second_sync_record_count_by_stream,
+         second_sync_state,
+         second_sync_all_records_by_stream) = self.make_connection_and_run_sync(
+             create_connection_kwargs={"original_properties": False},
+         )
+
+        all_metadata = self.expected_metadata()
+
+        for stream_name in self.expected_sync_streams():
+            stream_metadata = all_metadata[stream_name]
+            with self.subTest(stream=stream_name):
+
+                replication_method = stream_metadata.get(self.REPLICATION_METHOD)
+
+                first_sync_count = first_sync_record_count_by_stream.get(stream_name,0)
+                second_sync_count = second_sync_record_count_by_stream.get(stream_name,0)
+
+                self.first_sync_records = self.filter_output_file_for_records(
+                    first_sync_all_records_by_stream,
+                    stream_name
+                )
+
+                self.second_sync_records = self.filter_output_file_for_records(
+                    second_sync_all_records_by_stream,
+                    stream_name
+                )
+
+                if  replication_method == self.FULL_TABLE:
+                    """
+                    Testing Criteria:
+                    1. Verify that Sync B includes the same number of records as Sync A
+                    2. Verify the saved state following each sync does not contain a bookmark value
+                       for the stream.
+                    3. Verify that all records in Sync B are included in Sync A.
+                    """
+                    # Criteria 1
+                    self.assertEqual(first_sync_count, second_sync_count)
+
+                    # Criteria 2
+                    self.assertNotIn(stream_name, first_sync_state['bookmarks'])
+                    self.assertNotIn(stream_name, second_sync_state['bookmarks'])
+
+                    # Criteria 3
+                    first_sync_unique_records = self.get_unique_records(stream_name,
```

Comment:
I don't think this method should be used here. We are altering the sync results. I think it made sense how it was used in the pagination test but not here.